openpype/hosts/houdini/api/shelves.py

```diff
@@ -162,6 +173,7 @@ def get_or_create_tool(tool_definition, shelf):
         return
 
     script_path = tool_definition["script"]
+    script_path = get_path_using_template_data(script_path, template_data)
```

Comment:
is there any reason not to use ` shelf_set_os_filepath` variable in line 38 as part of the argument in this function instead of calling the function `get_path_using_template_data` twice?

Reply:
sure, you could pass `shelf_set_os_filepath` instead of `template_data` and save this call.

Reply:
I'm confused. 

all what I know that `script_path` is not the same as `shelf_set_os_filepath`

```python
shelf_set_os_filepath = project_settings["houdini"]["shelves"][shelf_index]["shelf_set_source_path"][current_os]
script_path = project_settings["houdini"]["shelves"][shelf_index]["shelf_definition"][definition_index]["tools_list"][tool_index]["script"]
```

also, 
if `shelf_set_os_filepath` has a value, code will ignore any tool definitions. 
if `shelf_set_os_filepath` is empty string `""`. code will look for script paths to assemble a shelf 

So, could you tell me what am I missing ?

Reply:
> I'm confused.
> 
> all what I know that `script_path` is not the same as `shelf_set_os_filepath`
> 
> ```python
> shelf_set_os_filepath = project_settings["houdini"]["shelves"][shelf_index]["shelf_set_source_path"][current_os]
> script_path = project_settings["houdini"]["shelves"][shelf_index]["shelf_definition"][definition_index]["tools_list"][tool_index]["script"]
> ```
> 
> also, if `shelf_set_os_filepath` has a value, code will ignore any tool definitions. if `shelf_set_os_filepath` is empty string `""`. code will look for script paths to assemble a shelf
> 
> So, could you tell me what am I missing ?

Ahhh....I can see why now...Ignore my comment then.